crates/core/tedge_mapper/src/aws/mapper.rs

```diff
@@ -52,7 +52,15 @@ impl TEdgeComponent for AwsMapper {
             );
             cloud_config.set_clean_session(false);
             cloud_config.set_keep_alive(aws_config.bridge.keepalive_interval.duration());
-            use_key_and_cert(&mut cloud_config, aws_config)?;
+
+            // set tls config
+            let cloud_broker_auth_config = tedge_config
+                .mqtt_auth_config_cloud_broker(aws_config)
+                .expect("error getting cloud broker auth config");
+            let client_config = cloud_broker_auth_config
+                .to_rustls_client_config()
+                .expect("error getting cloud broker auth config");
+            cloud_config.set_transport(Transport::tls_with_config(client_config.into()));
```

Comment:
This piece of code is repeated for c8y, aws and azure mappers. This will also be useful to let `tedge connect` validate a new certificate using a direct connection to the cloud.

Hence, I would add a new method in tedge config that returns a `rustls::ClientConfig`, to write:

```suggestion
            let tls_config =  tedge_config.rustls_client_config(aws_config)?;
            cloud_config.set_transport(Transport::tls_with_config(tls_config.into()));
```


Furthermore; the function `use_key_and_cert` is now only used by a test and can be safely removed.

Reply:
fixed in 884a5caa43